R/getEvents.R

```diff
@@ -65,27 +65,21 @@ getEvents <- function(experimentId,
   checkDefined(fcsFileId)
   fcsFileId <- lookupByName(paste("experiments", experimentId, "fcsfiles", sep = "/"), fcsFileId, "filename")
 
-  if (is.null(compensation) && !is.null(populationId)) {
-    stop("'compensation' parameter is required for gated populations.")
-  } else if (is.null(compensation) && isTRUE(compensatedQ)) {
-    stop("'compensation' parameter is required when returning compensated data.")
-  }
-
-  if (!is.null(subsampling$preSubsampleN) && !is.null(subsampling$preSubsampleP)) {
-    stop("Specify only one of preSubsampleN or preSubsampleP.")
-  }
+  checkArgs(compensation, compensatedQ, populationId, subsampling)
```

Comment:
I like the next commit (reducing complexity of `getStatistics`), but not a big fan of this one. The complexity here is only 16 and [the default limit was 25 until recently](https://github.com/jimhester/lintr/commit/ed393e7f1e1c5df9e2c9e3bdc5ea62caf58f9f1f#diff-8582f76c9ac94c6d6096f2e5b51542883ac02e853a7c0dd79b4791ccee5c9d0f). Could you change the linter to allow 25?

Alternatively, could you make them two separate functions: `checkCompensationArgs` and `checkSubsamplingArgs`?

Reply:
:+1: